### PR TITLE
Make layout compatible with upcoming jenkins core

### PR DESCRIPTION
**Pull request by timja**
As part of the UI refresh of Jenkins we are modernising it's layout so
it can be better styled, this plugin is identified as the only major
plugin that will break, these changes make it work on both old and new
core versions

Ref: https://github.com/jenkinsci/jenkins/pull/3895

Screenshots:
<details>

<summary>2.159</summary>

artifactory-2 159

</details>

<details>

<summary>2.227-SNAPSHOT</summary>

artifactory-2 227-SNAPSHOT

</details>

@jsoref
@eyalbe4

### CODE REVIEW

File: src/main/resources/org/jfrog/hudson/ArtifactoryBuilder/global.jelly [hunk 1]
```diff
@@ -2,21 +2,22 @@
...
     <st:adjunct includes="lib.jfrog.repos.credentials"/>
     <f:section title="Artifactory">
 
         <f:entry
             help="/plugin/artifactory/help/ArtifactoryBuilder/help-useCredentialsPlugin.html">
-                <input id="useCredentialsPlugin" type="checkbox" onclick="updateViewForCredentialsMethod(this.checked)"
-                   name="useCredentialsPlugin" value="${descriptor.useCredentialsPlugin}"
-                        field="useCredentialsPlugin">Use the Credentials Plugin
-                </input>
+                <f:checkbox id="useCredentialsPlugin"
```
Comment (timja) [inline src/main/resources/org/jfrog/hudson/ArtifactoryBuilder/global.jelly]: This isn't strictly related but it's an accessibility bug I noticed while testing out the UI

File: src/main/resources/org/jfrog/hudson/ArtifactoryBuilder/global.jelly [hunk 2]
```diff
@@ -2,21 +2,22 @@
          xmlns:f="/lib/form"
          xmlns:st="jelly:stapler"
          xmlns:r="/lib/jfrog"
-         xmlns:c="/lib/credentials">
+         xmlns:c="/lib/credentials" xmlns:l="/lib/layout">
```
Comment (timja) [inline src/main/resources/org/jfrog/hudson/ArtifactoryBuilder/global.jelly]:
```suggestion
         xmlns:c="/lib/credentials">
```

**Comment by timja** [comment]
@yahavi @eyalbe4 could you please take a look at this

**Comment by eyalbe4** [comment]
@timja,
Thanks for creating this pull request.

It looks like this change causes a UI issue:
With the current implementation, the **useCredentialsPlugin** toggles between two states:
**When checked** - the username and password are invisible and the credentials field is visible.
**When unchecked** - the username and password are visible and the credentials field is invisible.
Following your change, all fields are always visible. The reason for this is probably the fact that you removed the following two ids:
```
"resolverCredentialsId${server.url}"
"legacyResolverCredentials${server.url}"
```
These ids are used by the **toggleCredentialsPluginFromView** function inside **credentials.js**.

**Comment by timja** [comment]
Could you take a look at this please?

We plan to ship this in ~2 weeks in a weekly release after the next LTS cut-off

**Comment by eyalbe4** [comment]
@timja,
See my comment from a few minutes ago.
Thanks

**Comment by timja** [comment]
> @timja,
> See my comment from a few minutes ago.
> Thanks

please take another look now

**Comment by eyalbe4** [comment]
@timja,
The checkbox still doesn't affect the UI even after your last commits. This may be because you haven't restored all the IDs. 
Did this actually work for you? (testing this is really easy).

**Comment by timja** [comment]
> @timja,
> The checkbox still doesn't affect the UI even after your last commits. This may be because you haven't restored all the IDs.

It's not unfortunately (although it did mean that one of the toggles was broken but the other should have worked), there's some javascript which is moving the `f:entry` outside of the div, it's trying to keep the row location flat, but means it completely breaks trying to set an ID, and even more annoyingly none of the jelly tags allow you to pass an ID.

I see why the table tag is currently there as it's the only way I could get it to work without it being brittle to the layout of elements in the Jenkins UI, having the table there allows the next rows to be inside of the table rather than flat at the top level.

With the current state it works with the upcoming core version on the global config page, the highjacking doesn't seem to occur anymore.

> Did this actually work for you? (testing this is really easy).

I just merged in @jsoref's change, didn't test it myself


This can't be merged in the current state, but next steps depend on whether you would be willing to bump your core version when the weekly is released or whether you want to try stay compatible with both versions.

It should be easy to sort out on the new version (i.e. this appears to work fine for the global config page),
Working across both versions is going to be difficult

**Comment by timja** [comment]
Hmm I think to fix this without bumping core a taglib should be created that use j:choose / j:when and looks at the jelly property divBasedFormLayout and uses the appropriate tag either div or table

Should be straightforward, just needed to sleep on it

**Comment by timja** [comment]
> @timja,
> The checkbox still doesn't affect the UI even after your last commits. This may be because you haven't restored all the IDs.

resolved in https://github.com/jfrog/jenkins-artifactory-plugin/pull/266/commits/0fdce97ee7350b32e2bcce984273e22baa67b744

**Comment by eyalbe4** [comment]
Thank you so much @timja!
[HAP-1341](https://www.jfrog.com/jira/browse/HAP-1341) tracks this change.

**Pull request merged**